docs/client_connection_strategy.rst

```diff
@@ -69,13 +69,14 @@ The following are configuration element descriptions:
 - ``retry_max_backoff``: Specifies the upper limit for the backoff in
   seconds. Its default value is ``30``. It must be non-negative.
 - ``retry_multiplier``: Factor to multiply the backoff after a failed
-  retry. Its default value is ``1``. It must be greater than or equal
+  retry. Its default value is ``1.05``. It must be greater than or equal
```

Comment:
What happened to the old [`connection_attempt_limit `](http://hazelcast.github.io/hazelcast-python-client/3.10/hazelcast.config.html#hazelcast.config.ClientNetworkConfig.connection_attempt_limit) ?

Is there no longer an option to control the number of connection attempts?

Reply:
Hi, this configuration option is removed from the client in the 4.0 release with a more enhanced exponential backoff feature. You can get the almost same behavior by configuring the `cluster_connect_timeout` parameter.

Reply:
In my use case, I just want to either attempt 1 or 2 connection attempts and then stop (as part of an integration test).

The exponential backup is a nice feature (and it will be useful in production), but it looks like the logic for limiting the connection attempts is now somewhat convoluted.

Reply:
I can agree on some of your points but let me clarify the `cluster_connect_timeout` a little bit. 

It is actually an absolute timeout for whole connection attempts.

Let say that you specified 3 possible member addresses in your configuration. Client first will store the time it started trying to connect the cluster. (Let say that it is `start_time`)

Client tries to connect one of the three. If it cannot connect any of them, then client tries to sleep a little bit. But before sleeping, it will check the the time passed since the first connection attempt(`time.time() - start_time`). If it is more than the `cluster_connect_timeout`, it will throw an exception saying that client could not connect to any of the members. If it is less than that, it will sleep and try to connect possible addresses again. This loops continues until the client reaches the `cluster_connect_timeout` or connects one of the members.

Reply:
The same logic applies to reconnection process

Reply:
@mdumandag thanks for the quick response and clarification, that's very helpful.

Reply:
No problem at all, happy to help you. Let me rephrase that part tomorrow